src/conn/conn_api.c

```diff
@@ -1360,45 +1350,44 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[])
 }
 
 /*
- * __conn_write_config --
- *	Save the configuration used to create a database.
+ * __conn_write_base_config --
+ *	Save the base configuration used to create a database.
  */
 static int
-__conn_write_config(
-    WT_SESSION_IMPL *session, const char *filename, const char *cfg[])
+__conn_write_base_config(WT_SESSION_IMPL *session, const char *cfg[])
 {
 	FILE *fp;
 	WT_CONFIG parser;
-	WT_CONFIG_ITEM k, v;
+	WT_CONFIG_ITEM cval, k, v;
 	WT_DECL_RET;
-	char *path;
+	int exist;
+
+	fp = NULL;
 
 	/*
-	 * We were passed an array of configuration strings where slot 0 is all
-	 * possible values and the second and subsequent slots are changes
-	 * specified by the application during open (using the wiredtiger_open
-	 * configuration string, an environment variable, or user-configuration
-	 * file). The base configuration file contains all changes to default
-	 * settings made at create, and we include the user-configuration file
-	 * in that list, even though we don't expect it to change. Of course,
-	 * an application could leave that file as it is right now and not
-	 * remove a configuration we need, but applications can also guarantee
-	 * all database users specify consistent environment variables and
-	 * wiredtiger_open configuration arguments, and if we protect against
-	 * those problems, might as well include the application's configuration
-	 * file as well.
-	 *
-	 * If there is no configuration, don't bother creating an empty file.
+	 * Discard any base configuration setup file left-over from previous
+	 * runs.  This doesn't matter for correctness, it's just cleaning up
+	 * random files.
+	 */
+	WT_RET(__wt_remove_if_exists(session, WT_BASECONFIG_SET));
+
+	/* The base configuration file is optional, check the configuration. */
+	WT_RET(__wt_config_gets(session, cfg, "config_base", &cval));
+	if (!cval.val)
+		return (0);
+
+	/*
+	 * We don't test separately if we're creating the database as we might
+	 * have crashed between creating the "WiredTiger" file and creating the
+	 * base configuration file. There's always a base configuration file,
+	 * and we rename it into place, so it can only NOT exist if we crashed
+	 * before it was created. If it already exists, we're done.
```

Comment:
I've fixed the comment.

I tried to call this change out -- I agree with your analysis, and the two questions are first, if any harm could come from the change (I think the answer is "yes", but that harm is pretty unlikely), and if it would be a problem for Amazon or MongoDB. If it's not a problem for either of them, maybe it's OK.